lib/index.ts

```diff
@@ -454,6 +465,45 @@ export class DockerProgress {
 		return hash;
 	}
 
+	/** Build docker image calling onProgress with extended progress info regularly */
+	async build(
+		image: string,
+		onProgress: ProgressCallback,
+		options?: BuildOptions,
+	): Promise<string> {
+		// Create tar for build
+		const tarStream = tar.pack();
+		tarStream.entry({ name: 'Dockerfile' }, `FROM ${image}`, (err) => {
```

Comment:
This effectively hardcodes a Dockerfile in this function, even if just a single-line one. Given that `docker-progress` is a fairly generic package, I would suggest having the caller provide the Dockerfile instead. Furthermore, I would suggest that the caller actually provides `tarStream`, which could be any readable stream (not necessarily one produced with the `tar-stream` package). This would also avoid the need to add `tar-stream` to the `package.json` of `docker-progress`. The signature of the `build` method might then look like:

```
	async build(
		tarStream: NodeJS.ReadableStream,
		onProgress: ProgressCallback,
		options?: BuildOptions,
	): Promise<string> {
```

Or:

```
import type * as Stream from 'stream';

	async build(
		tarStream: Stream.Readable,
		onProgress: ProgressCallback,
		options?: BuildOptions,
	): Promise<string> {
```

I don't remember if `NodeJS.ReadableStream` is more/less generic or otherwise better/worse than `Stream.Readable`.

Reply:
Awesome suggestions @pdcastro! I'll get to work on the changes

Reply:
I like the idea of passing a `Stream` to the build function. This makes the `keepParentImage` option useless though. I think I'm OK with that given that removing the image is probably out of the scope of the function responsibilities.